File: app/frontend/common/types/ticket.ts
// Copyright (C) 2012-2022 Zammad Foundation, https://zammad-foundation.org/

enum TicketState {
  closed = 'closed',
  'waiting-for-closure' = 'waiting-for-closure',
  'waiting-for-reminder' = 'waiting-for-reminder',
  open = 'open',
  escalated = 'escalated',
}

export default TicketState

File: app/models/system_report/plugin/entities/ticket.rb
# Copyright (C) 2012-2024 Zammad Foundation, https://zammad-foundation.org/

class SystemReport::Plugin::Entities::Ticket < SystemReport::Plugin
  DESCRIPTION = __('Open and closed tickets ratio (ticket counts based on state)').freeze

  def fetch
    {
      'Open'   => Ticket.where(state_id: Ticket::State.by_category(:open)).count,
      'Closed' => Ticket.where(state_id: Ticket::State.by_category(:closed)).count,
    }
  end
end
